checkstyle-config.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+    <property name="severity" value="error"/>
+
+    <property name="fileExtensions" value="java, properties, xml"/>
+
+
+    <!--    This for ignoring specific module when start applying rules so cleaning rules violations would be easier and organized    -->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/models/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/core/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/common/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/integration/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/Agent/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/keploy-sdk/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/api/"/>-->
+    <!--    </module>-->
+    <!--    <module name="BeforeExecutionExclusionFileFilter">-->
+    <!--        <property name="fileNamePattern" value="/agent/"/>-->
+    <!--    </module>-->
```

Comment:
Where are the checks like - AvoidStarImport

Reply:
@charankamarapu 
I avoided to add rules right now to avoid pipeline failure
when I tried to run check it resulted +15K rule violation which won't be practical to solve in one issue and we don't have unit tests to to verify that everything still works as expected after such significant changes.

Reply:
Yeah I get it but u can use BeforeExecutionExclusionFileFilter - exclude all files and add all the rules so that there will be no errors. Then other developers will see the rules include each module and work on it . I mean If we don't specify rules what will we be benchmarking our code against? Please add the rules and exclude all the modules/files and also add a screenshot by including one module/file which shows all the errors.

Reply:
Sure on it